funsor/adjoint.py

```diff
@@ -13,8 +13,9 @@
 from funsor.interpreter import interpretation
 from funsor.ops import AssociativeOp
 from funsor.registry import KeyedRegistry
+from funsor.tensor_ops import Tensor, is_tensor, materialize
```

Comment:
I think `is_tensor` is a bad idea. Could we either use `isinstance(-, (torch.Tensor, numpy.ndarray))` or define a tuple
```py
# in tensor_ops
numeric_array = (torch.Tensor, numpy.ndarray)
```
to be used the same way we used to use `numbers.Number` back in Python 2?

Reply:
good idea! using `numeric_array` is simpler. I didn't know about this usage pattern. :D